### PR TITLE
fix(MdInput): Add a test and convert values entered based on the type of input

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

**Comment by jelbourn** [comment]
@hansl you accidentally git-added a bunch of output js files.

**Comment by hansl** [comment]
I blame WebStorm!

File: src/components/input/input.ts [hunk 1]
```diff
@@ -197,6 +203,19 @@ export class MdInput implements ControlValueAccessor, AfterContentInit, OnChange
   }
 
   /**
+   * Convert the value passed in to a value that is expected from the type of the md-input.
+   * This is normally performed by the *_VALUE_ACCESSOR in forms, but since the type is bound
+   * on our internal input it won't work locally.
+   * @private
+   */
+  private _convert(v: any): any {
```
Comment (jelbourn) [inline src/components/input/input.ts]: `_convertValueForInputType` ?

Comment (hansl) [inline src/components/input/input.ts]: donerino.

Comment (jelbourn) [inline src/components/input/input.ts]: I think you need a re-push.

Comment (hansl) [inline src/components/input/input.ts]: yeah apparently I did ><.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**